src/App.js

```diff
@@ -47,47 +50,55 @@ function App() {
               </PrivateAuth>
             }
           />
-          <Route path="/videolisting" element={<VideoListing />} />
-          <Route
-            path="/playlist"
-            element={
-              <RequiresAuth>
-                <Playlist />
-              </RequiresAuth>
-            }
-          />
-          <Route
-            path="/history"
-            element={
-              <RequiresAuth>
-                <History />
-              </RequiresAuth>
-            }
-          />
-          <Route
-            path="/profile"
-            element={
-              <RequiresAuth>
-                <Profile />
-              </RequiresAuth>
-            }
-          />
-          <Route
-            path="/watchlater"
-            element={
-              <RequiresAuth>
-                <WatchLater />
-              </RequiresAuth>
-            }
-          />
+          <Route path="/video" element={<Video />}>
+            <Route index path="videolisting" element={<VideoListing />} />
+            <Route path="videolisting/:videoId" element={<SingleVideo />} />
+            <Route
+              path="watchlater"
+              element={
+                <RequiresAuth>
+                  <WatchLater />
+                </RequiresAuth>
```

Comment:
You can explore having nested conditional routes, thus not having to do repetition

Reply:
OK, sure I will do the suggested changes